arangod/Aql/QueryRegistry.cpp

```diff
@@ -316,9 +367,18 @@ futures::Future<std::shared_ptr<ClusterQuery>> QueryRegistry::finishQuery(
     // last thread closes its engine
     return queryInfo._promise.getFuture();
   }
+  QueryDestructionContext queryDestructionContext(
+      queryInfo._queryString, queryInfo._errorCode, queryInfo._finished);
 
   auto result = queryInfo._query;
   deleteQuery(queryMapIt);
```

Comment:
This will still delete the `QueryInfo` under the lock. I think we want to play the same trick here as in `destroyQuery`.